tests/SideBySide.New/StoredProcedureTests.cs

```diff
@@ -0,0 +1,258 @@
+				});
+				cmd.Parameters.Add(new MySqlParameter
+				{
+					ParameterName = "@volume",
+					DbType = DbType.Double,
+					Direction = ParameterDirection.Output,
+				});
+				cmd.Parameters.Add(new MySqlParameter
+				{
+					ParameterName = "@shape",
+					DbType = DbType.String,
+					Direction = ParameterDirection.Output,
+				});
+
+				var result = await ExecuteCommandAsync(cmd, executorType);
+				if (executorType != "NonQuery")
+					Assert.Equal((string)cmd.Parameters["@name"].Value + (string)cmd.Parameters["@shape"].Value, result);
+
+				Assert.Equal(2 * (double)cmd.Parameters["@radius"].Value, cmd.Parameters["@diameter"].Value);
+				Assert.Equal(Math.PI * (double)cmd.Parameters["@radius"].Value, cmd.Parameters["@area"].Value);
+				Assert.Equal(Math.PI * Math.Pow((double)cmd.Parameters["@radius"].Value, 2), cmd.Parameters["@area"].Value);
+				Assert.Equal((double)cmd.Parameters["@area"].Value * (double)cmd.Parameters["@height"].Value, cmd.Parameters["@volume"].Value);
+			}
+		}
+
+		private async Task<object> ExecuteCommandAsync(DbCommand cmd, string executorType)
+		{
+			switch (executorType)
+			{
+				case "NonQuery":
+					await cmd.ExecuteNonQueryAsync();
+					return null;
+				case "Scalar":
+					return await cmd.ExecuteScalarAsync();
+				default:
+					using (var reader = await cmd.ExecuteReaderAsync())
+					{
+						if (await reader.ReadAsync())
+							return reader.GetValue(0);
+						return null;
+					}
+			}
+		}
+
+		[Theory]
+		[InlineData("@factor")]
+		[InlineData("factor")]
+		public async Task MultipleRows(string paramaterName)
+		{
+			using (var cmd = m_database.Connection.CreateCommand())
+			{
+				cmd.CommandText = @"
+				drop procedure if exists number_multiples;
+				drop table if exists sproc_multiple_rows;
+				create table sproc_multiple_rows (
+					value integer not null primary key auto_increment,
+					name text not null
+				);
+				insert into sproc_multiple_rows values
+				(1, 'one'),
+				(2, 'two'),
+				(3, 'three'),
+				(4, 'four'),
+				(5, 'five'),
+				(6, 'six'),
+				(7, 'seven'),
+				(8, 'eight');
+				create procedure number_multiples (in factor integer)
+				begin
+					select name from sproc_multiple_rows
+					where mod(value, factor) = 0
+					order by name;
+				end;";
+				await cmd.ExecuteNonQueryAsync();
+			}
+
+			using (var cmd = m_database.Connection.CreateCommand())
+			{
+				cmd.CommandText = "number_multiples";
+				cmd.CommandType = CommandType.StoredProcedure;
+				cmd.Parameters.Add(new MySqlParameter { ParameterName = paramaterName, Value = 3 });
+				using (var reader = await cmd.ExecuteReaderAsync())
+				{
+					Assert.True(await reader.ReadAsync());
+					Assert.Equal("six", reader.GetString(0));
+					Assert.True(await reader.ReadAsync());
+					Assert.Equal("three", reader.GetString(0));
+					Assert.False(await reader.ReadAsync());
+					//Assert.False(await reader.NextResultAsync()); // <-- This passes on MySql.Data but not MySqlConnector, why?
```

Comment:
There is. https://dev.mysql.com/doc/internals/en/multi-resultset.html says:

> ... and a closing empty resultset, an OK_Packet. ... The trailing OK_Packet is the response to the CALL statement and contains the affected-rows count of the last statement. 

I think the way to address this might be to add extra logic in `NextResultAsync` to detect this empty result set condition (when `CommandType == StoredProcedure`) and return an accurate `false` value.

I can think of one edge case where this will be a problem: someone using Dapper's `QueryMultiple` to call a sproc with multiple result sets. Their code will only read _n_ result sets but Dapper will think there are _n+1_, so will `Cancel` the command because it thinks there was an incomplete read. This will currently throw an exception in this library. (Dapper's regular `Query` for a sproc will work just fine.)

It seems fine to me to ship with this as a known issue and address it in a future version. (Shipping support for stored procedures may turn up other problems too, and we can fix them all at once.)